sos/report/plugins/__init__.py

```diff
@@ -2843,11 +2848,13 @@ def get_process_pids(self, process):
                 continue
         return pids
 
-    def get_network_namespaces(self, ns_pattern=None, ns_max=0):
+    def get_network_namespaces(self, ns_pattern=None, ns_max=None):
+        if ns_max is None and self.commons['cmdlineopts'].namespaces:
+            ns_max = self.commons['cmdlineopts'].namespaces
```

Comment:
Why the "`and self.commons[..]` option? If this evals to False, then the commons namespace is `None` or `0`. Either value can be assigned to `ns_max`, then, as "this allows the use of both '0' and `None` to mean unlimited".

Or do I skip some corner case..?

Reply:
Remember ns_max is the internally set limit. This allows users to override plugin settings of unlimited collections.

ns_max is `None` and commons is `None` -> ns_max is unlimited
ns_max is `None` and commons is `0` -> ns_max is unlimited, as the user intended by setting it to `0`
ns_max is `None` and commons is > `0` -> ns_max is limited, as the user intended
ns_max is not `None` and commons is `None` -> ns_max is limited by the call, the user never set an option value.

So it looks like yes there is a use case that is getting missed actually. Where ns_max is not None, and commons is set to something else. I'll get this fixed up.

Reply:
Oh, wait - duh. Sorry this escaped me at first since it's been a bit.

This is because of the per-plugin option to set a namespace limit. This allows users set individual plugin namespace limits separate from any global limit set. Using the plugin option is the only way `ns_max` has a non `None` value.